analyses/ssd/flood_trigger/ssd_floodscan.md

```diff
@@ -0,0 +1,372 @@
+    gdf_adm2.loc[gdf_adm2.ADM1_EN.isin(['Unity','Jonglei']),'geometry'])
+.hvplot.hist('SFED_AREA',alpha=0.5))
+```
+
+### Compute stats per county
+Now that we have analyzed the raster data, we can aggregate this to get the statistics per county. 
+
+```python tags=[]
+# #takes a bit of time so only compute when not saved yet
+# df_floodscan=compute_raster_statistics(
+#         gdf=gdf_adm2,
+#         bound_col='ADM2_PCODE',
+#         raster_array=da_clip,
+#         lon_coord="lon",
+#         lat_coord="lat",
+#         stats_list=["median","mean","max"],
+#         #computes value where 20% of the area is above that value
+#         percentile_list=[80],
+#         #Decided to only use centres, but can change that
+#         all_touched=False,
+#     )
+# df_floodscan=df_floodscan.merge(gdf_adm2[["ADM2_EN","ADM2_PCODE"]],on="ADM2_PCODE",how="left")
+# # df_floodscan.to_csv(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan_adm2_stats.nc',index=False)
+```
+
+```python
+df_floodscan=pd.read_csv(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan_adm2_stats.nc',parse_dates=['time'])
+```
+
+With these statistics we can plot the timeseries per admin2. 
+
+We plot these for the mean as well as for the median. We can see very different patterns, due to many cells having values around zero. I am not sure which method of aggregation makes most sense here.. I would incline towards the median though you then see that this drops very quickly again which might "downgrade" the presence of a flood.
+
+For both variables however, we can see that the patterns per county differ heavily. Both in absolute as in relative numbers. We do see clear peaks during some years, which is a good indication that this data and the aggregation methodology might at least be able to pick up fluctuations in the normal situation. 
+
+```python tags=[]
+def plot_floodscan_timeseries(df,var_col,adm_list,adm_col="ADM2_EN"):
+    for adm in adm_list:
+        fig, ax = plt.subplots()
+        df_floodscan_sel = df[df[adm_col]==adm].copy()
+        df_floodscan_sel.loc[:,'mean_cell_rolling'] = (df_floodscan_sel
+                                                       .loc[:,var_col]
+                                                       .transform(lambda x: x.rolling(5, 1).mean())
+                                                      )
+        sns.lineplot(data=df_floodscan_sel, x="time", y=var_col, lw=0.25, label='Original')
+        sns.lineplot(data=df_floodscan_sel, x="time", 
+                     y="mean_cell_rolling", lw=0.25, label='5-day moving\navg')   
+        ax.set_ylabel('Mean flooded fraction')
+        ax.set_xlabel('Date')
+        ax.set_title(f'Flooding in {adm}, 1998-2020, aggregation {var_col}')
+        ax.legend()
+```
+
+```python tags=[]
+plot_floodscan_timeseries(df_floodscan,"mean_ADM2_PCODE",['Fangak'])
+```
+
+```python
+plot_floodscan_timeseries(df_floodscan,"mean_ADM2_PCODE",adm2_list)
```

Comment:
The data from Leer looks very strange to me, how the baseline suddenly changes halfway through:
![download](https://user-images.githubusercontent.com/6585693/153367490-4176a9df-3249-4fb5-ab65-86aa74f96c58.png)

Also Twic East and Bor South are a bit weird, it's more difficult to see the seasonality:
![download](https://user-images.githubusercontent.com/6585693/153368687-72e9624b-1fdf-4735-b314-d6005573b0fd.png)
![download](https://user-images.githubusercontent.com/6585693/153368706-6253da83-5526-42ef-aef1-c0fb0a1bf015.png)

Reply:
Nice catches! Maybe we can ask the floodscan people about them.

Reply:
Reply from Floodscan on Leer: I haven’t found any obvious causes for the pattern in Leer. There are a few possibilities. An increase in vegetation cover over time might cause less flooding to be detected. However, I don’t see an obvious greenness trend over that time period. (I used NASA’s WorldView tool to browse through vegetation index in July each year when flooding is usual at its peak: https://worldview.earthdata.nasa.gov/). Changes in the sensors available to us might have an effect but those changes (in Oct. 2011, Oct. 2012, and April 2014) don’t seem to correlate to the trends in Leer either. Otherwise, I can only speculate. I suppose it’s possible that the floodplain evolved and silting or other factors reduced flooding in some areas for a time. More likely is that some of the area is on the margins of the seasonal flood and so has more extreme year-to-year variation in flood extent.